File: Nian_Karl/QL/src/org/uva/ql/ast/expression/binary/Or.java
package org.uva.ql.ast.expression.binary;

import org.uva.ql.ast.expression.Expression;
import org.uva.ql.ast.visitor.Visitor;

public class Or extends Binary {

	public Or(Expression left, Expression right) {
		super(left, right);
	}
	
	@Override
	public <T> T accept(Visitor<T> visitor) {
		return visitor.visit(this);
	}
	
	@Override
	public String toString() {
		return this.left.toString() + " || " + this.right.toString();
	}

}

File: FelipezConde/QuestionnaireLanguage/AST/Nodes/Expressions/Binary/Or.cs
using Expressions = AST.Nodes.Interfaces;
using AST.Representation;

namespace AST.Nodes.Expressions.Binary
{
    public class Or : Binary
    {
        public Or(Expression left, Expression right, PositionInText position)
         : base(left, right, position)
        { }

        public override T Accept<T>(ASTVisitors.IVisitor<T> visitor)
        {
            return visitor.Visit(this);
        }

        public override string ToString()
        {
            return "||";
        }
    }
}
